.gitignore

```diff
@@ -1 +1,32 @@
-.vscode/
\ No newline at end of file
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Compiled Static libraries
+*.lai
+*.la
+*.a
+*.lib
+
+# Executables
+*.exe
+*.out
+*.app
```

Comment:
Why do we need this?

Reply:
Again, these are possible files created or used by C++.  I figured it's better to be pro-active rather than reactive and piece meal lines into the .gitignore.

Reply:
It is easier to manage and ignore the files we know about and that we create vs just trying to block random file extensions we may or may not need.